rhasspywake_raven/dtw.py

```diff
@@ -90,32 +81,26 @@ def _compute_optimal_path(
         if len(y.shape) == 1:
             y = y.reshape(-1, 1)
 
-        distance_matrix = scipy.spatial.distance.cdist(x, y, metric=self.distance_func)
-
-        cost_matrix = np.full(shape=(m, n), fill_value=math.inf, dtype=float)
-        cost_matrix[0][0] = distance_matrix[0][0]
+        cost_matrix = scipy.spatial.distance.cdist(x, y, metric=self.distance_func)
 
         for row in range(1, m):
-            cost = distance_matrix[row, 0]
-            cost_matrix[row][0] = cost + cost_matrix[row - 1][0]
+            cost_matrix[row, 0] += cost_matrix[row - 1, 0]
 
         for col in range(1, n):
-            cost = distance_matrix[0, col]
-            cost_matrix[0][col] = cost + cost_matrix[0][col - 1]
+            cost_matrix[0, col] += cost_matrix[0, col - 1]
 
         for row in range(1, m):
             for col in range(1, n):
-                cost = distance_matrix[row, col]
-                cost_matrix[row][col] = cost + min(
-                    cost_matrix[row - 1][col],  # insertion
-                    cost_matrix[row][col - 1],  # deletion
-                    cost_matrix[row - 1][col - 1],  # match
+                cost_matrix[row, col] += min(
+                    cost_matrix[row - 1, col],  # insertion
+                    cost_matrix[row, col - 1],  # deletion
+                    cost_matrix[row - 1, col - 1],  # match
                 )
 
         if keep_matrix:
             self.cost_matrix = cost_matrix
 
-        distance = cost_matrix[m - 1][n - 1]
+        distance = cost_matrix[m - 1, n - 1]
```

Comment:
This should be faster (just a single call to a function). And especially it can be optimized automatically by cython later on